### PR TITLE
385 upgrade sqlalchemy

### LINKED ISSUE #385: Upgrade sqlalchemy
Now that Python has been upgraded, we can upgrade sqlalchemy to 2.0.

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes #385

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp/core/models/deployment_log.py [hunk 1]
```diff
@@ -74,41 +75,43 @@ class DeploymentLog(Base):
...
-    start_time = Column(DateTime(timezone=False))
-    end_time = Column(DateTime(timezone=False))
-    state: Column = Column(Enum(DeploymentStateEnum))
-    deployment_type: Column = Column(Enum(DeploymentTypeEnum))
-    restart = Column(Boolean, default=False)
-
-    operations = relationship(
-        "OperationLog",
+    id: Mapped[int] = mapped_column(primary_key=True)
+    sources: Mapped[Optional[list]] = mapped_column(JSON(none_as_null=True))
```
Comment (rpignolet) [inline tdp/core/models/deployment_log.py:95]: You can add the type for item in the list or it does not work ?
```suggestion
    sources: Mapped[Optional[list[str]]] = mapped_column(JSON(none_as_null=True))
```

Comment (PaulFarault) [reply tdp/core/models/deployment_log.py:95]: Yes I can!

Comment (PaulFarault) [reply tdp/core/models/deployment_log.py:95]: Done

**Review by rpignolet — APPROVED** [review]

**Comment by mdrutel** [comment]
When I execute ```tdp deploy --mock-deploy``` after ```tdp plan dag```, I have the followinf warning :
```
/home/mdrutel01/projets/git/tdp-ops/tdp/tdp-lib/tdp/cli/commands/deploy.py:79: SAWarning: Object of type <ComponentVersionLog> not in session, add operation along 'DeploymentLog.component_version' will not proceed (This warning originated from the Session 'autoflush' process, which was invoked automatically in response to a user-initiated operation.)
  session.merge(operation_log)
```

**Review by rpignolet — APPROVED** [review]

**Review by mdrutel — APPROVED** [review]
all the tests I did were OK :ok_hand:

**Pull request merged**